docs/decisions/0003-django-setting-toggles.rst

```diff
@@ -32,41 +32,29 @@ We will introduce a simple wrapping class (or classes) that can be used to expli
 Here are some example instances::
 
     # Toggle setting
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingToggle("EXAMPLE_TOGGLE", default=False)
+    EXAMPLE_SETTING_TOGGLE = SettingToggle("EXAMPLE_TOGGLE", default=False)
 
     # or
 
-    # Toggle setting in a dict: Option 1
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingToggle(
-        "EXAMPLE_SETTING_DICT['EXAMPLE_TOGGLE']",
-        default=False,
-    )
-
-    # or
-
-    # Toggle setting in a dict: Option 2
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingDictToggle(
+    # Toggle setting in a dict
+    EXAMPLE_SETTING_TOGGLE = SettingDictToggle(
         'EXAMPLE_SETTING_DICT',
         'EXAMPLE_TOGGLE',
         default=False,
     )
 
-    # TODO: Before implementing, we'll need to choose between Option 1 and Option 2 for
-    #   toggle setting in a dict. We can update this decision once this has been decided,
-    #   and move the rejected option, explaining why.
-
 It's usage can now match that of the waffle wrapping classes (e.g. WaffleFlag).
 
 Here is an example usage::
 
     EXAMPLE_SETTING_TOGGLE.is_enabled()
 
-DjangoSettingToggle will allow for expicit toggle definitions, with the following benefits:
+SettingToggle will allow for explicit toggle definitions, with the following benefits:
 
 * The definition can live with the app where it will be used, rather than in `common.py` or with no explicit definition.
 * The explicit definition can be annotated with documentation.
 * The explicit definition provides a single place to set a consistent default value.
-* The `toggle state endpoint`_ can provide the location and code ownership of all DjangoSettingToggle instances used in an enviroment, because the class will track its instances.
+* The `toggle state endpoint`_ can provide the location and code ownership of all SettingToggle instances used in an enviroment, because the class will track its instances.
```

Comment:
We'll likely end up with situations where we go through a SettingToggle in some places and directly to Django settings in others, which seems fine. However, we may also end up with multiple SettingToggle declarations for the same setting. I suppose the class can warn us about that?

Reply:
1. We could lint for setting usage that doesn't use SettingToggle, if they are including a default like True/False, to detect it is a toggle.
2. SettingToggle could detect if multiple declarations happen for the same toggle, but I don't expect that will be a big issue.